src/main/java/org/jabref/gui/slr/ExistingStudySearchAction.java

```diff
@@ -38,19 +39,26 @@
     private final FileUpdateMonitor fileUpdateMonitor;
     private final TaskExecutor taskExecutor;
     private final PreferencesService preferencesService;
-    private final ImportFormatPreferences importFormatPreferneces;
+    private final StateManager stateManager;
+    private final ImportFormatPreferences importFormatPreferences;
     private final SavePreferences savePreferences;
     // This can be either populated before crawl is called or is populated in the call using the directory dialog. This is helpful if the directory is selected in a previous dialog/UI element
 
-    public ExistingStudySearchAction(JabRefFrame frame, FileUpdateMonitor fileUpdateMonitor, TaskExecutor taskExecutor, PreferencesService preferencesService) {
+    public ExistingStudySearchAction(JabRefFrame frame,
+                                     FileUpdateMonitor fileUpdateMonitor,
+                                     TaskExecutor taskExecutor,
+                                     PreferencesService preferencesService,
+                                     StateManager stateManager) {
         this.frame = frame;
         this.dialogService = frame.getDialogService();
         this.fileUpdateMonitor = fileUpdateMonitor;
-        this.workingDirectory = getInitialDirectory(preferencesService.getWorkingDir());
         this.taskExecutor = taskExecutor;
         this.preferencesService = preferencesService;
-        this.importFormatPreferneces = preferencesService.getImportFormatPreferences();
+        this.stateManager = stateManager;
+        this.importFormatPreferences = preferencesService.getImportFormatPreferences();
```

Comment:
Why keep both preferencesService and the smaller preferences objects?

Reply:
As a temporary placeholder on the way to use only the small objects.